modules/apps/transfer/keeper/grpc_query_test.go

```diff
@@ -140,3 +140,71 @@ func (suite *KeeperTestSuite) TestQueryParams() {
 	res, _ := suite.queryClient.Params(ctx, &types.QueryParamsRequest{})
 	suite.Require().Equal(&expParams, res.Params)
 }
+
+func (suite *KeeperTestSuite) TestQueryDenomHash() {
+
+	reqTrace := types.DenomTrace{
+		Path:      "transfer/channelToA/transfer/channelToB",
+		BaseDenom: "uatom",
+	}
+
+	var (
+		req     *types.QueryDenomHashRequest
+		expHash = reqTrace.Hash().String()
+	)
+
+	testCases := []struct {
+		msg      string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"invalid trace",
+			func() {
+				req = &types.QueryDenomHashRequest{
+					Trace: "transfer/channelToA/transfer/",
+				}
+			},
+			false,
+		},
+		{
+			"not found denom trace",
+			func() {
+				req = &types.QueryDenomHashRequest{
+					Trace: reqTrace.GetFullDenomPath(),
+				}
+			},
+			false,
+		},
+		{
+			"success",
+			func() {
+				suite.chainA.GetSimApp().TransferKeeper.SetDenomTrace(suite.chainA.GetContext(), reqTrace)
+
+				req = &types.QueryDenomHashRequest{
+					Trace: reqTrace.GetFullDenomPath(),
+				}
+			},
+			true,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
+			suite.SetupTest() // reset
+
+			tc.malleate()
+			ctx := sdk.WrapSDKContext(suite.chainA.GetContext())
+
+			res, err := suite.queryClient.DenomHash(ctx, req)
+
+			if tc.expPass {
```

Comment:
```suggestion
	testCases := []struct {
		msg      string
		malleate func()
		expPass  bool
	}{
		{
			"invalid trace",
			func() {
				req = &types.QueryDenomHashRequest{
					Trace: "transfer/channelToA/transfer/",
				}
			},
			false,
		},
		{
			"not found denom trace",
			func() {
				req = &types.QueryDenomHashRequest{
					Trace: "transfer/channel-5000/atom",
				}
			},
			false,
		},
		{
			"success",
			func() {},
			true,
		},
	}

	for _, tc := range testCases {
		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
			suite.SetupTest() // reset

                        req = &types.QueryDenomHashRequest{
					Trace: reqTrace.GetFullDenomPath(),
			}
			suite.chainA.GetSimApp().TransferKeeper.SetDenomTrace(suite.chainA.GetContext(), reqTrace)
			
			tc.malleate()
			ctx := sdk.WrapSDKContext(suite.chainA.GetContext())

			res, err := suite.queryClient.DenomHash(ctx, req)

			if tc.expPass {
```

Reply:
@colin-axner I have applied your test code suggestions, thank you for your patience with me!